Setup.hs

```diff
@@ -50,12 +60,19 @@ genBuildInfo verbosity pkg info = do
     ]
   rewriteFile targetText buildVersion
 
-timestamp :: IO String
-timestamp =
-  formatTime defaultTimeLocale "%Y%m%d%H%M%S" `fmap` getCurrentTime
-
 gitVersion :: IO String
 gitVersion = do
   ver <- readProcess "git" ["log", "--pretty=format:%h", "-n", "1"] ""
   notModified <- ((>) 1 . length) `fmap` readProcess "git" ["status", "--porcelain"] ""
   return $ ver ++ if notModified then "" else "-M"
+
+timestamp :: IO String
+timestamp =
+  readProcess "date" ["+%Y%m%d%H%M%S"] "" >>= \s ->
+    case splitAt 14 s of
+      (d, n : []) ->
+        if (length d == 14 && filter isDigit d == d)
+          then return d
+          else fail $ "date has failed to produce the correct format [" <> s <> "]."
+      _ ->
+        fail $ "date has failed to produce a date long enough [" <> s <> "]."
```

Comment:
nope, too risky for the moment

cabal-1.24 adds support for Setup.hs dependency lists